ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectIndex.cpp

```diff
@@ -67,6 +67,10 @@ api_index_aspect(PyObject* self, PyObject* const* args, const Py_ssize_t nargs)
         return nullptr;
     }
 
+    if (args == nullptr) {
+        return nullptr;
```

Comment:
I would ask for some comment explaining the rationale behind this

Reply:
At least for a regular string index, Python raises a `TypeError` instead of just returning `None`:
```
>>> "fff"[None]
<stdin>:1: SyntaxWarning: str indices must be integers or slices, not NoneType; perhaps you missed a comma?
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string indices must be integers
```

Reply:
- The checking: pointers should always be checked for nullptr before using them.
- The `return nullptr`: for exception propagation to work, you have to return `nullptr`. The CPython API won't check if there is an exception if you don't return `nullptr`.